internal/controller/instaslice_controller_test.go

```diff
@@ -44,6 +44,83 @@ import (
 	"github.com/openshift/instaslice-operator/internal/controller/utils"
 )
 
+var _ = Describe("NodeReconciler", func() {
+	var (
+		ctx        context.Context
+		scheme     *runtime.Scheme
+		rNode      *NodeReconciler
+		node       *v1.Node
+		instaslice *inferencev1alpha1.Instaslice
+		objs       []client.Object
+	)
+
+	BeforeEach(func() {
+		ctx = context.TODO()
+		scheme = runtime.NewScheme()
+		_ = v1.AddToScheme(scheme)
+		_ = inferencev1alpha1.AddToScheme(scheme)
+		objs = []client.Object{}
+	})
+
+	newFakeClient := func() client.Client {
+		return fake.NewClientBuilder().WithScheme(scheme).WithObjects(objs...).Build()
+	}
+
+	Describe("NodeReconciler", func() {
+		BeforeEach(func() {
+			node = &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "fake-node",
+					Labels: map[string]string{
+						"nvidia.com/mig.capable":  "true",
+						"nvidia.com/mig.strategy": "mixed",
+					},
+				},
+				Status: v1.NodeStatus{
+					Capacity:    v1.ResourceList{},
+					Allocatable: v1.ResourceList{},
+					NodeInfo:    v1.NodeSystemInfo{BootID: "boot-123"},
+				},
+			}
+			objs = append(objs, node)
+		})
+
+		It("should do nothing if node is labeled as managed", func() {
+			node.Labels = map[string]string{
+				"instaslice.redhat.com/managed": "true",
+			}
+			rNode = &NodeReconciler{Client: newFakeClient()}
+			req := reconcile.Request{NamespacedName: types.NamespacedName{Name: node.Name}}
+			result, err := rNode.Reconcile(ctx, req)
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
Just checking for error is not sufficient. Please also make sure the instaslice object is not deleted by mistake.